paasta_tools/kubernetes/bin/paasta_secrets_sync.py

```diff
@@ -396,6 +434,86 @@ def sync_secrets(
     return True
 
 
+def sync_database_credentials(
+    kube_client: KubeClient,
+    cluster: str,
+    service: str,
+    secret_provider_name: str,
+    vault_cluster_config: Dict[str, str],
+    soa_dir: str,
+    vault_token_file: str,
+) -> bool:
+    config_loader = PaastaServiceConfigLoader(service=service, soa_dir=soa_dir)
+    system_paasta_config = load_system_paasta_config()
+    database_credentials_vault_overrides = (
+        system_paasta_config.get_database_credentials_vault_overrides()
+    )
+
+    for instance_config in config_loader.instance_configs(
+        cluster=cluster, instance_type_class=KubernetesDeploymentConfig
+    ):
+        database_credentials = instance_config.get_database_credentials()
+        with set_env(**database_credentials_vault_overrides):
+            # expects VAULT_ADDR_OVERRIDE, VAULT_CA_OVERRIDE, and VAULT_TOKEN_OVERRIDE to be set
```

Comment:
This can go away too if you move it to its own cron job? Or that plus a small change to vault_tools to support non standard vault shards?

Reply:
> move to its own cron job
definitely possible if we set the environment variables to the scope of the bash command, though @nemacysts advised this may enter some confusing territory when running manually and advised against it: https://github.com/Yelp/paasta/pull/3655#discussion_r1268381304

> a small change to vault_tools
unfortunately this wouldn't be small by my estimates. if you go through the vault_tools/paasta_secret.py file, you'll see the concept of ecosystem baked in everywhere and hardcoded paths and naming conventions. for fear of doing this incorrectly, we opted to use the overrides, which is a feature of the library anyway:
https://github.com/Yelp/paasta/pull/3655#discussion_r1272745254

Reply:
++ I'm sure vault_tools will need to be refactored at some point if more teams start requesting custom vault shards. I filed an onpoint ticket with PE-auto for something unrelated and highlighted the outdated docs

Reply:
@mattmb perhaps we could refactor this file a bit and move all the helper functions for each secret type to their own file? that would leave the existing file as just the driver/entrypoint for everything?

we might want to do that in a separate PR so that this diff doesn't become impossible to read tho :p

Reply:
asked them again: https://yelp.slack.com/archives/CCYF30A1K/p1691092164669369

think that should be sufficient for resolving this thread - feel free